File: GanttChartLightLibraryDemos/Demos/Samples/WPF-CSharp/GanttChartDataGrid/Recurrence/UnlimitedIntConverter.cs
using System;
using System.Windows.Data;
using System.Globalization;

namespace Demos.WPF.CSharp.GanttChartDataGrid.Recurrence
{
    public class UnlimitedIntConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
        {
            int intValue = (int)value;
            return intValue < int.MaxValue ? intValue.ToString() : "Unlimited";
        }

        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
        {
            string stringValue = (string)value;
            if (stringValue == string.Empty || stringValue.ToLowerInvariant() == "unlimited")
                return int.MaxValue;
            stringValue = stringValue.ToLowerInvariant().Replace("unlimited", string.Empty);
            int intValue;
            if (int.TryParse(stringValue, out intValue))
                return intValue;
            return 0;
        }
    }
}

File: GanttChartLightLibraryDemos/Demos/Samples.Resources/WPF-VisualBasic/GanttChartDataGrid/Recurrence/UnlimitedIntConverter.vb
Imports System.Globalization

Public Class UnlimitedIntConverter
    Implements IValueConverter

    Public Function Convert(value As Object, targetType As Type, parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.Convert
        Dim intValue As Integer = CInt(Fix(value))
        Return If(intValue < Integer.MaxValue, intValue.ToString(), "Unlimited")
    End Function

    Public Function ConvertBack(value As Object, targetType As Type, parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.ConvertBack
        Dim stringValue As String = CStr(value)
        If stringValue = String.Empty OrElse stringValue.ToLowerInvariant() = "unlimited" Then
            Return Integer.MaxValue
        End If
        stringValue = stringValue.ToLowerInvariant().Replace("unlimited", String.Empty)
        Dim intValue As Integer
        If Integer.TryParse(stringValue, intValue) Then
            Return intValue
        End If
        Return 0
    End Function
End Class
